### PR TITLE
[WEB-1680] dev: issue detail activity revamp and issue detail page improvement

### PR DESCRIPTION
**Pull request by anmolsinghbhatia**
#### Changes:
This PR includes the following updates:
- Revamped the issue detail activity.
- Updated the order of issue widgets in peek overview.
- Improved padding on the issue detail page.
- Fixed the attachment delete modal issue where clicking on delete was closing the peek overview.
- Resolved the relation issue delete toast alert problem, which was previously rendering two toast alerts.

#### Media:
| Activity |
|--------|
| media |

| Peek overview |
|--------|
| media |

### CODE REVIEW

File: web/core/components/issues/attachment/attachment-list-item.tsx [hunk 1]
```diff
@@ -33,10 +33,16 @@ export const IssueAttachmentsListItem: FC<TIssueAttachmentsListItem> = observer(
   const { getUserDetails } = useMember();
   const {
     attachment: { getAttachmentById },
+    toggleAttachmentDeleteModal,
   } = useIssueDetail();
   // state
   const [isDeleteIssueAttachmentModalOpen, setIsDeleteIssueAttachmentModalOpen] = useState(false);
```
Comment (SatishGandham) [inline web/core/components/issues/attachment/attachment-list-item.tsx:39]: if you are maintaining modal state in store, why are we having it here again?

File: web/core/components/issues/issue-detail/issue-activity/activity-filter.tsx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+type Props = {
+  selectedFilters: EActivityFilterType[];
+  toggleFilter: (filter: EActivityFilterType) => void;
+};
+
+export const ActivityFilter: FC<Props> = observer((props) => {
+  const { selectedFilters, toggleFilter } = props;
+  return (
+    <Popover as="div" className="relative">
```
Comment (SatishGandham) [inline web/core/components/issues/issue-detail/issue-activity/activity-filter.tsx:20]: We have a component for this now. Please use it.

File: web/core/components/issues/issue-detail/issue-activity/activity-comment-root.tsx [hunk 3]
```diff
@@ -28,24 +31,34 @@ export const IssueActivityCommentRoot: FC<TIssueActivityCommentRoot> = observer(
...
-            ends={index === 0 ? "top" : index === activityComments.length - 1 ? "bottom" : undefined}
+            ends={index === 0 ? "top" : index === filteredActivityComments.length - 1 ? "bottom" : undefined}
             showAccessSpecifier={showAccessSpecifier}
             disabled={disabled}
           />
         ) : activityComment.activity_type === "ACTIVITY" ? (
           <IssueActivityList
             activityId={activityComment.id}
-            ends={index === 0 ? "top" : index === activityComments.length - 1 ? "bottom" : undefined}
+            ends={index === 0 ? "top" : index === filteredActivityComments.length - 1 ? "bottom" : undefined}
```
Comment (SatishGandham) [inline web/core/components/issues/issue-detail/issue-activity/activity-comment-root.tsx:61]: What is ends?

File: web/core/components/issues/issue-detail/issue-activity/activity-comment-root.tsx [hunk 4]
```diff
@@ -28,24 +31,34 @@ export const IssueActivityCommentRoot: FC<TIssueActivityCommentRoot> = observer(
...
             workspaceSlug={workspaceSlug}
             commentId={activityComment.id}
             activityOperations={activityOperations}
-            ends={index === 0 ? "top" : index === activityComments.length - 1 ? "bottom" : undefined}
+            ends={index === 0 ? "top" : index === filteredActivityComments.length - 1 ? "bottom" : undefined}
             showAccessSpecifier={showAccessSpecifier}
             disabled={disabled}
           />
         ) : activityComment.activity_type === "ACTIVITY" ? (
           <IssueActivityList
```
Comment (SatishGandham) [inline web/core/components/issues/issue-detail/issue-activity/activity-comment-root.tsx:59]: Why is this not issueActivityCard

File: web/core/components/issues/attachment/attachment-list-item.tsx (hunk 1 above)
Comment (anmolsinghbhatia) [reply web/core/components/issues/attachment/attachment-list-item.tsx:39]: Resolved

File: web/core/components/issues/issue-detail/issue-activity/activity-comment-root.tsx (hunk 4 above)
Comment (anmolsinghbhatia) [reply web/core/components/issues/issue-detail/issue-activity/activity-comment-root.tsx:59]: Resolved

File: web/core/components/issues/issue-detail/issue-activity/activity-filter.tsx (hunk 2 above)
Comment (anmolsinghbhatia) [reply web/core/components/issues/issue-detail/issue-activity/activity-filter.tsx:20]: That component is not flexible for use here, so I need to create a global filter popover component. I have created a ticket WEB-1894 for this task and will take care of it.

File: web/core/components/issues/issue-detail/issue-activity/activity-comment-root.tsx (hunk 3 above)
Comment (anmolsinghbhatia) [reply web/core/components/issues/issue-detail/issue-activity/activity-comment-root.tsx:61]: A ticket has been raised for this WEB-1895. I'll take care of it.

**Review by SatishGandham — APPROVED** [review]

**Pull request merged**